test/physics/HostDebugSecondaryStorage.cc

```diff
@@ -0,0 +1,32 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file HostDebugSecondaryStorage.cc
+//---------------------------------------------------------------------------//
+#include "HostDebugSecondaryStorage.hh"
+
+namespace celeritas_test
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Resize and give unassigned valuess to the secondaries.
+ *
+ * The values here are to help ensure that the secondary allocator is properly
+ * initializing the data when it operates.
+ */
+void HostDebugSecondaryStorage::resize(size_type capacity)
+{
+    Secondary unallocated_secondary;
+    unallocated_secondary.def_id    = celeritas::ParticleDefId{0xdeadbeef};
```

Comment:
This is great until someone in particle physics decides that a particle id of 3735928559 is absolutely needed 🤣

Reply:
In all seriousness, I would probably prefer `0xffffffff` although `deadbeef` has the benefit of automatically assuming you have 8 niblets in the number.  Perhaps, just (-1) which should get cast using the default opaque id unsigned type?

Reply:
This is basically  a  test that  `SecondaryAllocatorView::operator()` does placement `new` on the returned result. This value won't show up otherwise even to test harness code. Both `0` and `-1` (undefined) would be valid values, so something grossly out of bounds for a unit test is needed, and I chose a [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)#Magic_debug_values) used for indicating uninitialized memory.